lorawan: make unconfirmed message type explicit

martinjaeger opened the pull request:
The enum did not contain an entry for unconfirmed messages. Instead, it was only mentioned in the comment that 0 is the default for unconfirmed messages.

This PR adds `LORAWAN_MSG_UNCONFIRMED` to the enum and clarifies some comments in the documentation.

martinjaeger committed:
lorawan: make unconfirmed message type explicit

The enum did not contain an entry for unconfirmed messages. Instead, it
was only mentioned in the comment that 0 is the default for
unconfirmed messages.

This commit adds LORAWAN_MSG_UNCONFIRMED to the enum and changes the
parameter in the lorawan_send function to enum lorawan_message_type.

Signed-off-by: Martin Jäger <martin@libre.solar>

JordanYates commented on include/lorawan/lorawan.h:
```diff
@@ -202,13 +201,12 @@ int lorawan_start(void);
  * @return 0 if successful, negative errno code if failure
  */
-int lorawan_send(uint8_t port, uint8_t *data, uint8_t len, uint8_t flags);
+int lorawan_send(uint8_t port, uint8_t *data, uint8_t len, uint8_t type);
```
If you're changing `could be one of the lorawan_message_type` to `Must be one of lorawan_message_type` then you may as well make the type parameter explicit `enum lorawan_message_type type`

martinjaeger replied:
The idea was that existing code doesn't break, but everyone who reads the docs does it correct.

I'm also fine with `enum lorawan_message_type`. Let's see what others think.

Mani-Sadhasivam replied:
During the initial API review we decided not to define a type for unconfirmed message as it is the default. But I think defining it also doesn't hurt but then please change the datatype of `type` to enum.

Mani-Sadhasivam replied:
Oh wait... Please let the `flags` parameter stay as it is. We may want to add some flags in the future.

Mani-Sadhasivam replied:
> Are there any other types than confirmed and unconfirmed?
> 
> If we want to use these flags for something else than the message type in the future, the enum should be called `lorawan_message_flags` instead. Otherwise there is no relation to the different bits that can be set as flags.

After going back and forth, I came to the conclusion that we should be just using `lorawan_message_type`. So please change the datatype of the `type` parameter.

Mani-Sadhasivam commented on include/lorawan/lorawan.h:
```diff
@@ -202,13 +201,12 @@ int lorawan_start(void);
- *                   could be one of the lorawan_message_type. The default
- *                   behaviour is unconfirmed message.
+ * @param type       Specifies if the message shall be confirmed or unconfirmed.
+ *                   Must be one of lorawan_message_type.
```
I don't think we should specifying the enum types here.

martinjaeger committed:
doc: release-notes: mention API change for lorawan_send

The message type parameter was changed to the more explicit enum.
Existing code with unconfirmed message type has to be updated.

Signed-off-by: Martin Jäger <martin@libre.solar>

martinjaeger commented:
@mniestroj can you please also have a look?
